sessionize/app/src/main/java/co/touchlab/sessionize/NotificationPublisher.kt

```diff
@@ -0,0 +1,79 @@
+package co.touchlab.sessionize
+
+import android.app.Activity
+import android.app.AlarmManager
+import android.app.NotificationManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.app.Notification
+import android.app.PendingIntent
+import android.os.Build
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import co.touchlab.sessionize.platform.AndroidAppContext
+
+class NotificationPublisher : BroadcastReceiver() {
+
+    companion object {
+        const val NOTIFICATION_ID = "notificationId"
+        const val NOTIFICATION_TAG = "notificationTag"
+
+        const val NOTIFICATION_TITLE = "notificationTitle"
+        const val NOTIFICATION_MESSAGE = "notificationMessage"
+        const val NOTIFICATION_CHANNEL_ID = "notificationChannelID"
+
+        const val NOTIFICATION_ACTION_CREATE = "notificationCreate"
+        const val NOTIFICATION_ACTION_DISMISS = "notificationDismiss"
+    }
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        context?.let {
```

Comment:
100% optional:
I tend to dislike early returns, but I sorta feel like 
`if (context == null || intent == null) return` 
is a bit cleaner than nesting like this. I'm conflicted though and welcome someone disagreeing